### PR TITLE
Add dbs_account service

### PR DESCRIPTION
**Pull request by Andrei-Masilevich**

### CODE REVIEW

File: libraries/chain/dbs_account.cpp [hunk 1]
```diff
@@ -0,0 +1,524 @@
...
+    FC_CAPTURE_AND_RETHROW((name))
+}
+
+void dbs_account::check_account_existence(const account_name_type& name,
+                                          const optional<const char*>& context_type_name) const
+{
+    auto acc = db_impl().find<account_object, by_name>(name);
+    if (context_type_name.valid())
+    {
+        FC_ASSERT(acc != nullptr, "\"${b}\" \"${b}\" must exist.", ("a", *context_type_name)("b", name));
```
Comment (abolinbsu) [inline libraries/chain/dbs_account.cpp]: Change to $1 and $2, fix bug

File: libraries/chain/include/scorum/chain/dbs_base_impl.hpp [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    explicit dbservice_dbs_factory(database&);
+
+    virtual ~dbservice_dbs_factory();
+
+public:
+    template <typename ConcreteService> ConcreteService& obtain_service()
+    {
+        auto it = _dbs.find(typeid(ConcreteService).name());
+        if (BOOST_UNLIKELY(it == _dbs.end()))
```
Comment (abolinbsu) [inline libraries/chain/include/scorum/chain/dbs_base_impl.hpp]: I would delete BOOST_UNLIKELY

File: libraries/chain/scorum_evaluator.cpp [hunk 3]
```diff
@@ -43,43 +46,13 @@ inline void validate_permlink_0_1(const string& permlink)
...
-        case '2':
-        case '3':
-        case '4':
-        case '5':
-        case '6':
-        case '7':
-        case '8':
-        case '9':
+        // clang-format off
+        case 'a': case 'b': case 'c': case 'd': case 'e': case 'f': case 'g': case 'h': case 'i':
```
Comment (abolinbsu) [inline libraries/chain/scorum_evaluator.cpp]: change to
if (!std::islower(c) && !std::isdigit(c) && !(c == '-'))
             FC_ASSERT(false, "Invalid permlink character: ${s}", ("s", std::string() + c));

File: libraries/chain/scorum_evaluator.cpp [hunk 4]
```diff
@@ -43,43 +46,13 @@ inline void validate_permlink_0_1(const string& permlink)
...
+        case 'j': case 'k': case 'l': case 'm': case 'n': case 'o': case 'p': case 'q': case 'r':
+        case 's': case 't': case 'u': case 'v': case 'w': case 'x': case 'y': case 'z': case '0':
+        case '1': case '2': case '3': case '4': case '5': case '6': case '7': case '8': case '9':
         case '-':
+            // clang-format on
             break;
         default:
             FC_ASSERT(false, "Invalid permlink character: ${s}", ("s", std::string() + c));
```
Comment (abolinbsu) [inline libraries/chain/scorum_evaluator.cpp]: Use constructor

File: libraries/chain/scorum_evaluator.cpp [hunk 5]
```diff
@@ -97,24 +70,26 @@ struct strcmp_equal
 
 void witness_update_evaluator::do_apply(const witness_update_operation& o)
 {
-    _db.get_account(o.owner); // verify owner exists
+    dbs_account& accountService = _db.obtain_service<dbs_account>();
```
Comment (alexplevako) [inline libraries/chain/scorum_evaluator.cpp]: I would prefer to use same variable naming conversion around the project. That means that variables should be in lower case and splitted with _.

File: libraries/chain/dbs_account.cpp (hunk 1 above)
Comment (alexplevako) [reply libraries/chain/dbs_account.cpp]: I suppose it should be ${a} and ${b}

**Pull request merged**